### DISCUSSION TITLE
Resizing - is it me or is it a limitation in hs or lua?

### DISCUSSION THREAD

**Discussion by applebait** [Hammerspoon/hammerspoon#3561] [Q&A]
I'm trying to replicate the workflow I have in other tools but in Hammerspoon. One of them is resizing by making a window larger or smaller, having its pivot in the window's center. I have it but if I hold down the keys the window starts to dance around, the same doesn't happen if I go step by step, I mean, releasing one of the keys in each step.
Here's the code I'm using:

```

function changeWindowSize(delta)
    local win = hs.window.focusedWindow()
    local f = win:frame()

    -- calculate new window size
    local w = f.w + delta
    local h = f.h + delta

    -- keep the window centered
    f.x = f.x - delta / 2
    f.y = f.y - delta / 2
    f.w = w
    f.h = h

    win:setFrame(f)
end

-- Watch for key-down events
watcher1 = hs.eventtap.new({hs.eventtap.event.types.keyDown}, function(event)
    local flags = event:getFlags()
    if flags.ctrl and flags.alt then
        local character = event:getCharacters(true)
        if character == '-' then
            changeWindowSize(-10)
        elseif character == '=' then
            changeWindowSize(10)
        end
    end
end)

watcher1:start()

```

Please, let me know if it's me or an Hammerspoon/Lua limitation.

Thanks

**Comment by piechologist** [comment]
Why don't you use the hotkey module? It's much easier:
```lua
hs.hotkey.bind("ctrl-alt", "-", "Make window smaller", function() changeWindowSize(-10) end)
hs.hotkey.bind("ctrl-alt", "=", "Make window larger", function() changeWindowSize(10) end)
```

**Reply by applebait** [reply]
Sorry! You mean replacing the entire code block I shared with these 2 lines?

**Comment by piechologist** [comment]
Yes, the two lines replace the whole `watcher1` part. Just keep your `changeWindowSize` function. 

As a bonus tipp, try:
```lua
hs.hotkey.showHotkeys("ctrl-alt", "/")
```
That shows an overlay with all your defined shortcuts while holding Ctrl-Alt-/ down (or whatever binding you choose). Docs are [here](https://www.hammerspoon.org/docs/hs.hotkey.html).
